core/src/macros.rs

```diff
@@ -397,6 +399,9 @@ macro_rules! godot_wrap_method {
 
                 use std::cell::RefCell;
                 use std::panic::{self, AssertUnwindSafe};
+                use $crate::GodotObject;
+
+                let $owner: $owner_ty = <$type_name as $crate::NativeClass>::Base::from_sys(this);
```

Comment:
Maybe this should use `cast<$owner_ty>` instead? That would mean the owner type can be an up- (or even down-) cast and does not necessarilty have to match?

The use-case for that is probably very low, and since right now casts are quite expensive I'm not sure if that would be worth it to add to *every single function call*.